service/service.go

```diff
@@ -78,6 +80,22 @@ func New(config Config) (*Service, error) {
 
 	var err error
 
+	features := make(map[string]bool)
+	{
+		// FeatureGates contains a comma separated list like "MachinePool=true,AnotherFeature=false"
```

Comment:
I believe using a separate parameter for each feature would be easier to maintain. (so we would access each one by using `config.Flag.Service.FeatureGates.MachinePool`, for example)

What do you think?

Reply:
No strong feelings about it. It may be simpler that way. I did it like this because it's how k8s does it.